app/src/main/java/com/example/musicapp/ui/songdetails/SongDetailFragment.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.musicapp.ui.songdetails
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import com.example.musicapp.R
+import com.example.musicapp.controller.MediaController
+import com.example.musicapp.data.model.Song
+import com.example.musicapp.data.source.local.dao.ResolverDataImp
+import com.example.musicapp.data.source.repository.MusicRepository
+import com.example.musicapp.ui.main.MainActivity
+import com.example.musicapp.ui.notification.SongService
+import kotlinx.android.synthetic.main.fragment_detail_song.*
+
+class SongDetailFragment : Fragment(R.layout.fragment_detail_song), SongContract.View {
+    private var songAdapter: SongAdapter? = null
+    private var songPresenter: SongPresenter? = null
+    private var indexSong = -1
+    private var songs: MutableList<Song> = mutableListOf()
+    private var serviceSong: SongService? = null
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        initView()
+    }
+
+    private fun initView() {
+        songAdapter = SongAdapter()
+        songPresenter = context?.let { ResolverDataImp(it) }?.let {
+            MusicRepository(it)
+        }?.let {
+            SongPresenter(
+                this,
+                it
+            )
+        }
```

Comment:
Đoạn này được coi là lạm dùng scope function vì nó bị nhập nhằng, nhiều it quá 
Trong trường hợp này em nên khai báo để rõ ràng hơn 
```suggestion
        val resolver = context?.let(ResolverDataImp::new)?: return 
        val repository = MusicRepository(resolver)
        songPresenter =   SongPresenter(this, repository)
```
Tham khảo thêm 
![image](https://user-images.githubusercontent.com/53286739/106077133-8eb71b80-6143-11eb-906b-49f3f1ba6fc3.png)